main/src/ca/uwaterloo/flix/language/phase/Safety.scala

```diff
@@ -117,6 +146,69 @@ object Safety {
     }
   }
 
+  /**
+    * Returns `true` if the given `defn` has the public modifier.
+    */
+  private def isPub(defn: Def): Boolean = {
+    defn.spec.mod.isPublic
+  }
+
+  /**
+    * Returns `true` if the given `defn` has a polymorphic type.
+    */
+  private def isPolymorphic(defn: Def): Boolean = {
+    defn.spec.tparams.nonEmpty
+  }
+
+  /**
+    * Returns `Nil` if the given `defn` has exportable types according to
+    * [[isExportType]], otherwise a list of the found errors is returned.
+    */
+  private def checkExportableTypes(defn: Def): List[SafetyError] = {
+    val types = defn.spec.fparams.map(_.tpe) :+ defn.spec.retTpe
+    types.flatMap{ t =>
+      if (isExportType(t)) Nil
+      else List(SafetyError.IllegalExportSignature(t.loc, s"Exported function can only use simple types, not `$t`"))
+    }
+  }
+
+  /**
+    * Returns `true` if the given `tpe` is supported in exported signatures.
+    *
+    * Currently this is only types that are immune to erasure, i.e. the
+    * primitive java types and Object.
+    *
+    * Note that signatures should be exportable as-is, this means that even
+    * though the signature contains `MyAlias[Bool]` where
+    * `type alias MyAlias[t] = t`, it uses flix-specific type information which
+    * is not allowed.
+    */
+  private def isExportType(tpe: Type): Boolean = tpe.typeConstructor match {
+    case None => false
+    case Some(value) => value match {
+      case TypeConstructor.Bool => true
+      case TypeConstructor.Char => true
+      case TypeConstructor.Float32 => true
+      case TypeConstructor.Float64 => true
+      case TypeConstructor.Int8 => true
+      case TypeConstructor.Int16 => true
+      case TypeConstructor.Int32 => true
+      case TypeConstructor.Int64 => true
+      case TypeConstructor.BigInt => true
```

Comment:
String? :)

Reply:
String is erased to Object, so the signature is not unchanged. This would require actual infrastructure